WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionFilters.js

```diff
@@ -0,0 +1,662 @@
+    }));
+  };
+
+  const editSelectedEvents = (eventId) => {
+    setSelectedEvents((prevSelectedEvents) => (
+      prevSelectedEvents.includes(eventId)
+        ? prevSelectedEvents.filter((id) => id !== eventId)
+        : [...prevSelectedEvents, eventId]
+    ));
+  };
+  const editSelectedRegion = (region) => {
+    setSelectedRegion(region === 'all' ? '' : region);
+    setCompetitionApiKey((prevKey) => ({
+      ...prevKey,
+      region: region === 'all' ? '' : region,
+    }));
+  };
+  const editSelectedDelegate = (delegateId) => {
+    setSelectedDelegate(delegateId === 'None' ? '' : delegateId);
+    setCompetitionApiKey((prevKey) => ({
+      ...prevKey,
+      delegate: delegateId === 'None' ? '' : delegateId,
+    }));
+  };
+  const editPastSelectedYear = (newYear) => {
+    setPastSelectedYear(newYear);
+    if (timeOrder === 'past') {
+      setCompetitionApiKey((prevKey) => ({ ...prevKey, sort_by: 'past', year: newYear }));
+    }
+  };
+  const editCustomStartDate = (date) => {
+    setCustomStartDate(date);
+    setCompetitionApiKey((prevKey) => ({
+      ...prevKey,
+      sort_by: 'custom',
+      year: `start${date?.toISOString().split('T')[0] || ''}
+      end${customEndDate?.toISOString().split('T')[0] || ''}`,
+    }));
+  };
+  const editCustomEndDate = (date) => {
+    setCustomEndDate(date);
+    setCompetitionApiKey((prevKey) => ({
+      ...prevKey,
+      sort_by: 'custom',
+      year: `start${customStartDate?.toISOString().split('T')[0] || ''}
+      end${date?.toISOString().split('T')[0] || ''}`,
+    }));
+  };
+  const editSearchQuery = (q) => {
+    setSearchQuery(q);
+    setCompetitionApiKey((prevKey) => ({
+      ...prevKey,
+      search: q,
+    }));
+  };
+  const editTimeOrder = (order) => {
+    if (order === 'past') {
+      setCompetitionApiKey((prevKey) => ({ ...prevKey, sort_by: 'past', year: pastSelectedYear }));
+    } else if (order === 'custom') {
+      // Calling this in a separate function somehow avoids the problem with setState
+      // being asynchronous and causing the query key to be strangely changed back and forth
+      updateCustomCompetitionApiKey();
+    } else {
+      setCompetitionApiKey((prevKey) => ({ ...prevKey, sort_by: order, year: '' }));
+    }
+
+    setTimeOrder(order);
+  };
+
+  const [inProgressComps, setInProgressComps] = useState([]);
+  const [notInProgressFutureComps, setNotInProgressFutureComps] = useState([]);
+  const [recentComps, setRecentComps] = useState([]);
+  const [sortByAnnouncementComps, setSortByAnnouncementComps] = useState([]);
+  const [pastComps, setPastComps] = useState({});
+  const [customDatesComps, setCustomDatesComps] = useState([]);
+  const [mapDisplayComps, setMapDisplayComps] = useState([]);
+
+  const editPastComps = (comps, year) => {
+    setPastComps((prevPastComps) => ({
+      ...prevPastComps,
+      [year]: comps,
+    }));
+  };
+
+  const {
+    data: competitionsData,
```

Comment:
How does infinite query work - `data` has a `pages` property, which is an array of competition data, and fetching the next page pushes to the end of `data.pages`?

Reply:
Sounds correct